conans/client/manager.py

```diff
@@ -318,7 +318,7 @@ def install(self, reference, current_path, remote=None, options=None, settings=N
         if manifest_manager:
             manifest_manager.print_log()
 
-    def _load_deps_info(self, current_path, conanfile, output):
+    def _load_deps_info(self, current_path, conanfile, output, for_imports=False):
```

Comment:
Don't like the for_imports parameter, bad name (for_imports talks about the external use) and big if. Refactor and try again.